01-ruby-fundamentals/07-oo-classes.md

```diff
@@ -17,14 +18,61 @@ Array.new
 # => []
 ```
 
-Where we would normally use the _string literal_ `""`, we can also call the `.new` method on the `String` class to create a new string object. A class can be identified by it's __capitalized first letter__. So `""`, `[]`, and `{}` are syntactic sugar; using _literals_ to _initialize_ a new _instance_ of an _object type_. Most classes do **NOT** have this kind of convenience.
+A class can be identified by it's __capitalized first letter__. In the above snippet `String`, `Hash` and `Array` are all classes. When you call `.new` on a class, you get an _instance_ of that class. In any program, there may be zero, one or many instances of any given class.
+
+While we normally use _string literals_ like `""` or `"elephant"` to create a new string, we can also call the `.new` method on the `String` class to create a new string object. `""`, `[]`, and `{}` are syntactic sugar; using _literals_ to _initialize_ a new _instance_ of an _object type_. Most classes do **NOT** have this kind of convenience.
+
+### State and Behavior
+
+Objects are used to tie together _state_ and _behavior_. State is any data or attributes the object needs to keep track of, and behavior is methods that rely on or modify the state.
+
+Let's look at `String` again. A `String` definitely keeps track of some data: the letters in the string! This is the _state_.
+
+`String`s also have many methods that you can call, like `upcase` or `split`. This is the _behavior_. A key observation is that the same method called on different strings will produce different results - `"elephant".upcase` is not the same as `"hotdog".upcase`. Even though the procedure to be followed is the same, the method behaves differently depending on the object's state.
+
+In Ruby, state will be represented using _attributes_, also known as _instance variables_, and behavior will be defined using _methods_.
+
+### What Are Objects Used For?
+
+It can be intimidating to look at `String`, `Hash` or `Array` when learning about objects for the first time. These are fundamental programming concepts that get used in almost every program, how can we ever hope to build or reason about something as complex as that?
+
+Although any abstract idea can be represented as an object, it's sometimes easy to think of a physical object to get the idea. Let's imagine a program designed to keep track of a library catalogue, which might have a class called `Book`. A book has _state_, like `title`, `author`, and `copies_checked_out`. Some of this changes frequently and some of it will never change, but it's still all state. A book in our catalogue also has _behavior_, like `check_in`, `check_out`, or `place_hold`.
```

Comment:
Can this be turned into a small brainstorm question so they have to think about this themselves?